### PR TITLE
186811096 - remove old validator

### PR DESCRIPTION
**Pull request by kamtarnowski**
#### What's this PR do?
- Modifies logic saving validator during CLI Solana request (#validators_info_save)

#### How should this be manually tested?
- `rails test`
1)
- find a validator with network 'testnet' and account '2eCPrXeWo9Cg79cK6eyJdaCoiMDKJMPq7sAhXSP3spQk'. Change its attribute (e.g. name, keybase_id, www_url) and save it.
- run `rails r script/validators_get_info.rb`
- check if the validator attributes are valid (current name is "Edgevana Validator")
2)
- run rails console and remove the validator (account: '2eCPrXeWo9Cg79cK6eyJdaCoiMDKJMPq7sAhXSP3spQk')
- run `rails r script/validators_get_info.rb` again
- in rails console try to find the validator ('2eCPrXeWo9Cg79cK6eyJdaCoiMDKJMPq7sAhXSP3spQk'). It should not exist.
3)
- create a validator with relations assigned
```
      validator_histories
      validator_block_histories
      vote_accounts
      commission_histories
      stake_account_histories
      stake_accounts
      validator_ips
      validator_score_v1
      user_watchlist_elements
```
- run `script/one_time_scripts/remove_validator.rb {validator.network} {validator.account}`
- the validator should be completely erased with its all relations.

#### What are the relevant tickets?
[- URL to the PivotalTracker ticket](https://www.pivotaltracker.com/story/show/186811096)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

File: script/one_time_scripts/remove_validator.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  validator.commission_histories.delete_all
+  validator.stake_account_histories.delete_all
+  validator.stake_accounts.delete_all
+  validator.validator_ips.delete_all
+  validator.validator_score_v1&.delete
+  validator.user_watchlist_elements.delete_all
+  validator.delete
+
+  unless Rails.env.test?
+    puts "validator_histories deleted"
```
Comment (kbiala) [inline script/one_time_scripts/remove_validator.rb:33]: The whole point of these messages was to see the progress. This way it makes no sense. Please revert it.

File: test/logic/solana_logic_test.rb [hunk 2]
```diff
@@ -269,17 +269,28 @@ def validators_info
     end
   end
 
-  test 'validators_info_save' do
+  test '#validators_info_save updates persisted validators' do
```
Comment (kbiala) [inline test/logic/solana_logic_test.rb:272]:
```suggestion
  test '#validators_info_save updates data for existing validators' do
```

File: test/logic/solana_logic_test.rb [hunk 3]
```diff
@@ -269,17 +269,28 @@ def validators_info
...
     )
 
+    create(
+      :validator,
+      :mainnet,
+      name: 'test_name',
+      keybase_id: 'test_keybase_id',
+      details: 'test_details',
+      avatar_url: 'test_avatar_url',
+      www_url: 'test_www_url',
```
Comment (kbiala) [inline test/logic/solana_logic_test.rb:285]: To simulate real life example you should create a validator without name, keybase_id, details, avatar etc. These data should be completed after script finishes.

**Review by kbiala — CHANGES_REQUESTED** [review]

File: script/one_time_scripts/remove_validator.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+@network ||= ARGV[0]
+@account ||= ARGV[1]
+
+# raise "invalid network - #{network}" unless NETWORKS.include? network
```
[the remark this answers is not in the document]
Comment (kamtarnowski) [reply script/one_time_scripts/remove_validator.rb]: It was unintentional probably because of doing some tests.

Added account checker.

**Review by kbiala — APPROVED** [review]

**Pull request merged**